### PR TITLE
add automatic Black as a pre-commit event

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR add automatic Black as a pre-commit event

**How to prepare for test**:

**How to test**:
- [x] checkout branch 
- [x] make a commit 

**Expected test outcome**:
- [x] black runns on the commit
- [x] Take a screenshot and attach or copy/paste the output.

**How to test CI**:
- [x] open the travis-PR-build

**Expected test outcome**:
- [x] `cg` production dependencies section should be there and green.
- [x] `cg` development dependencies section should be there and green.
- [x] `cg` production dependencies section should not be there.
- [x] `cg` black section should be red (since cg is not yet Black formatted).

**Review:**
- [x] code approved by @ingkebil 
- [x] tests executed by @patrikgrenfeldt 
- [x] developers informed on Operations meeting by @patrikgrenfeldt 
- [x] mentioned in the cg/README.md  
- [x] "Merge and deploy" approved by @ingkebil 
Thanks for filling in who performed the code review and the test!

This is minor **version bump** because it adds some dev func. without altering the end user func of the project

### CODE REVIEW

**Comment by patrikgrenfeldt** [comment]
Test result:
image

**Comment by ingkebil** [comment]
This needs to be documented in the devman as well training needs to be
given so everyone is aware of this nice feature.

**Comment by patrikgrenfeldt** [comment]
There's a need to do 'pre-commit install' to get the actual feature though,
maybe we could add that in a dev-section in setup.py?

Den mån 9 dec. 2019 20:55Kenny Billiau <notifications@github.com> skrev:

**Comment by patrikgrenfeldt** [comment]
Closing PR since it became to messy with the reverts, the few actual changes are made in #499

**Comment by patrikgrenfeldt** [comment]
closed the wrong one :)

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ python:
 
 # command to install dependencies
```
Comment (ingkebil) [inline .travis.yml]:
```suggestion
# command to install dependencies
# pip locked to version in P_main
```

File: .travis.yml [hunk 2]
```diff
@@ -16,17 +16,28 @@ cache: pip
...
+    - name: "production dependencies"
       script:
         - pip check
+    - name: "development dependencies"
+        script:
+          - pip install -U -r requirements.txt -r requirements-dev.txt .
+          - pip check
     - name: "coverage"
       script:
+        - pip install --upgrade pip
```
Comment (ingkebil) [inline .travis.yml]: Just making sure I'm understanding this correctly: coveralls requires to have the latest pip installed?

File: .travis.yml [hunk 3]
```diff
@@ -16,17 +16,28 @@ cache: pip
...
         - pip install -U -r requirements-dev.txt .
         - coverage run --branch --omit="*.html" --source cg setup.py test
         - coveralls
+    - name: "code formatting"
+      script:
+        - pip install --upgrade pip
```
Comment (ingkebil) [inline .travis.yml]: Same here: black requires the latest pip?

**Comment by patrikgrenfeldt** [comment]
Test results: 
image
No "Dependecies" section 👍

File: .travis.yml [hunk 4]
```diff
@@ -16,17 +16,28 @@ cache: pip
...
+        - pip install black
+        - black --check cg tests
     - name: "linting"
       script:
-        - pip install -r requirements-dev.txt
+        - pip install --upgrade pip
```
Comment (ingkebil) [inline .travis.yml]: Same here: linting requires the latest version of pip?

File: .travis.yml (hunk 2 above)
Comment (patrikgrenfeldt) [reply .travis.yml]: Not that I know of no, I wouldn't mind upgrading it, especially now when we downgrade it to the production version.. What are your thoughts?

File: .travis.yml (hunk 3 above)
Comment (patrikgrenfeldt) [reply .travis.yml]: no

File: .travis.yml (hunk 4 above)
Comment (patrikgrenfeldt) [reply .travis.yml]: no

**Comment by patrikgrenfeldt** [comment]
Test results:
image
Production dependencies 👍 
Development dependencies 👍 
Code formatting 👍

File: .travis.yml (hunk 2 above)
Comment (patrikgrenfeldt) [reply .travis.yml]: As we talked about IRL, to get back up to the current PIP version assuming it's both the latest and greatest

**Pull request merged**